woocommerce/wc-admin
# Fix SelectControl focus and de-focus bug

Fixes #6879

- Fixes `<SelectControl>` component from unintended selecting the first option when focusing and de-focusing (such as navigating forms via Tab/Shift-Tab).
- Also changes the behaviour to automatically scroll to the currently selected option when the options list is displayed.

### Accessibility




### Screenshots

Automatic scroll:

#### Before:
<img src="https://user-images.githubusercontent.com/3747241/116519746-6edef180-a904-11eb-88ba-c7237e19bcae.png" width="400">

#### After:
<img src="https://user-images.githubusercontent.com/3747241/116520508-676c1800-a905-11eb-9159-a6e6eeeede1c.png" width="400">


### Detailed test instructions:

1. Navigate to Setup Wizard.
2. Select any country in `Country / Region` except for the first option (Afghanistan).
3. Set focus to `Address line 2` by clicking on the text field.
4. Press `Tab` twice, so that the Country field is focused and de-focused.
5. Observe the selected option did not change to Afghanistan.
